File: CSharp/Forms/Examples/RichTextBox/RichTextBox.cs
using System;
using System.Windows.Forms;

namespace RichTextBoxExample {
  class Form1 : Form {
    public Form1() {
      this.StartPosition = FormStartPosition.Manual;
      this.Text = "RichTextBox example";
      this.Location = new System.Drawing.Point(200, 200);
      this.ClientSize = new System.Drawing.Size(640, 480);

      this.text.Parent = this;
      this.text.Font = new System.Drawing.Font(System.Drawing.FontFamily.GenericMonospace, 10);
      System.Windows.Forms.Screen[] screens = System.Windows.Forms.Screen.AllScreens;
      for (int i = 0; i < screens.Length; ++i)
        this.text.Text += string.Format("Device {0} :\n  - Primary = {1}\n  - Name = {2}\n  - Screen = {3}\n  - Area = {4}\n\n", i, screens[i].Primary, screens[i].DeviceName, screens[i].Bounds, screens[i].WorkingArea);
      this.text.Bounds = new System.Drawing.Rectangle(0, 0, this.Width, this.Height);
      this.text.Anchor = AnchorStyles.Left | AnchorStyles.Top | AnchorStyles.Right | AnchorStyles.Bottom;
    }

    private RichTextBox text = new RichTextBox();
    }

  class MainClass {
    [STAThread]
    public static void Main() {
      Application.EnableVisualStyles();
      Application.Run(new Form1());
    }
  }
}

File: Examples/Forms/RichTextBox/Sources/RichTextBox.cpp
#include <Pcf/Pcf>

using namespace System;
using namespace System::Windows::FormsD;

namespace RichTextBoxExample {
  class Form1 : public Form {
  public:
    // The main entry point for the application.
    static void Main() {
      Application::EnableVisualStyles();
      Application::Run(Form1());
    }
    
    Form1() {
      this->Text = "RichTextBox example";
      this->StartPosition = FormStartPosition::Manual;
      this->Location = System::Drawing::Point(200, 200);
      this->ClientSize = System::Drawing::Size(640, 480);
      
      this->text.Parent = *this;
      this->text.Font = System::Drawing::Font(System::Drawing::FontFamily::GenericMonospace);
      System::Array<System::Windows::FormsD::Screen> screens = System::Windows::FormsD::Screen::AllScreens;
      for (int i = 0; i < screens.Length; ++i)
        this->text.Text += string::Format("Device {0} :\n  - Primary = {1}\n  - Name = {2}\n  - Screen = {3}\n  - Area = {4}\n\n", i, screens[i].Primary, screens[i].DeviceName, screens[i].Bounds, screens[i].WorkingArea);
      this->text.Bounds = System::Drawing::Rectangle(0, 0, this->Width, this->Height);
      this->text.Anchor = AnchorStyles::Left | AnchorStyles::Top | AnchorStyles::Right | AnchorStyles::Bottom;
    }
    
  private:
    RichTextBox text;
  };
}

pcf_startup (RichTextBoxExample::Form1)
